### PR TITLE
Remove Python 2 configuration and targets

### PR DESCRIPTION
**Pull request by rickeylev**
Bazel is dropping support for Python 2

Also raises the minimum Bazel version in the examples to 5.0, as 4.0 has built-in dependencies on Python 2.

Fixes https://github.com/bazelbuild/rules_foreign_cc/issues/1005

Work towards https://github.com/bazelbuild/bazel/issues/17293

### CODE REVIEW

**Comment by rickeylev** [comment]
re: CI failure: The failing case is the "minimum supported version test" because @bazel_tools//tools/android:strip_resources wants to use Python 2.

That was fixed back in 2019: https://github.com/bazelbuild/bazel/commit/d93a146463e20f4eedc5ae658b03ee0260750aa7

As far as I can tell, that code is in the 4.0 release. So I'm not sure what's going on.

File: .bazelci/config.yaml [hunk 1]
```diff
@@ -201,18 +201,13 @@ tasks:
       - "--experimental_enable_aggregating_middleman=False"
   min_supported_version_examples:
     name: "Minimum Supported Version Examples"
-    bazel: "4.0.0"
```
Comment (jsharpe) [inline .bazelci/config.yaml:204]: Technically bazel 4 is still under LTS support until the end of this year and so I'd rather that we just disable the failing tests and keep this at 4 as the rest of these changes only affect the examples / tests and don't change the ability of this ruleset to continue support v4.

**Review by jsharpe — CHANGES_REQUESTED** [review]
Thanks for the fixes but can you just disable the failing tests rather than bump the minimum supported version?

File: .bazelci/config.yaml (hunk 1 above)
Comment (rickeylev) [reply .bazelci/config.yaml]: I restored 4.0 as the min version.

I'll try my hand at modifying the config.yaml to skip the offending tests, but I might come back asking for some help on how to do so. (If you happen to have a link that explains these bazelci config files, that'd be great; the only thing i've found is https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/README.md which doesn't explain many of the things i see or how to use them).

Comment (rickeylev) [reply .bazelci/config.yaml]: OK, I think I figured it out.

Ready for another look

**Review by jsharpe — APPROVED** [review]
LGTM

**Pull request merged**